pkg/apis/hive/v1/validating-webhooks/machinepool_validating_admission_hook.go

```diff
@@ -284,6 +285,11 @@ func validateMachinePoolSpecInvariants(spec *hivev1.MachinePoolSpec, fldPath *fi
 		allErrs = append(allErrs, validateOpenStackMachinePoolPlatformInvariants(p, platformPath.Child("openstack"))...)
 		numberOfMachineSets = 1 // no zone support on OpenStack
 	}
+	if p := spec.Platform.VSphere; p != nil {
+		platforms = append(platforms, "vsphere")
+		allErrs = append(allErrs, validateVSphereMachinePoolPlatformInvariants(p, platformPath.Child("vsphere"))...)
+		numberOfMachineSets = 1 // no zone support on vSphere
```

Comment:
I would prefer `numberOfMachineSets` to be 0 so that the error when `spec.Autoscaling.MinReplicas` is less than 1 is
```
minimum replicas must at least 1
```
instead of
```
minimum replicas must be at least the number of zones
```
particularly since vSphere does not have zones.

Reply:
Should this be changed for OpenStack as well?

@dgoodwin thoughts as well?

I'll change both if you guys agree.

Reply:
Yes, it should be changed for OpenStack as well.